### PR TITLE
Clear window before draw BFW-890 

### PR DESCRIPTION
**Pull request by espr14**

### CODE REVIEW

File: src/gui/dialogs/DialogStateful.hpp [hunk 1]
```diff
@@ -83,12 +78,12 @@ class IDialogStateful : public IDialog {
     bool Change(uint8_t phase, uint8_t progress_tot, uint8_t progress); // = 0; todo should be pure virtual
     virtual ~IDialogStateful();
 
-    static constexpr rect_ui16_t get_radio_button_size() {
-        rect_ui16_t rc_btn = { 0, 32, 240, 320 - 96 }; //msg box size
-        rc_btn.y += (rc_btn.h - 40);                   // 30pixels for button (+ 10 space for grey frame)
-        rc_btn.h = 30;
-        rc_btn.x += 6;
-        rc_btn.w -= 12;
+    static rect_ui16_t get_radio_button_size() {                                // cannot be const(expr)
```
Comment (RadekVana) [inline src/gui/dialogs/DialogStateful.hpp:81]: It should be constexpr
problem is gui_defaults is written in C and will be rewritten to C++

Comment (RadekVana) [reply src/gui/dialogs/DialogStateful.hpp:81]: I recommend modifying gui_defaults to have `#ifdef __cplusplus` version

Comment (espr14) [reply src/gui/dialogs/DialogStateful.hpp:78]: I would leave it for different PR focused on C -> C++ transition.

Comment (espr14) [reply src/gui/dialogs/DialogStateful.hpp:78]: BFW-954 was made.

Comment (espr14) [reply src/gui/dialogs/DialogStateful.hpp:78]: However, it's duplicit work to review this before 889 is merged.

File: src/gui/dialogs/DialogStateful.cpp [hunk 2]
```diff
@@ -2,6 +2,11 @@
 #include "DialogRadioButton.hpp"
 #include "gui.h"
 
+constexpr uint8_t PROGRESS_BAR_X_PAD = 10;
```
Comment (RadekVana) [inline src/gui/dialogs/DialogStateful.cpp:5]: should be
`constexpr static const` or `constexpr static` (equal in this case)

Comment (espr14) [reply src/gui/dialogs/DialogStateful.cpp:5]: constexpr const == constexpr
I will add static.

File: src/gui/dialogs/DialogStateful.hpp [hunk 3]
```diff
@@ -80,12 +75,12 @@ class IDialogStateful : public IDialog {
     bool Change(uint8_t phs, uint8_t progress_tot, uint8_t progress); // = 0; todo should be pure virtual
     virtual ~IDialogStateful();
 
-    static constexpr rect_ui16_t get_radio_button_size() {
-        rect_ui16_t rc_btn = { 0, 32, 240, 320 - 96 }; //msg box size
-        rc_btn.y += (rc_btn.h - 40);                   // 30pixels for button (+ 10 space for grey frame)
-        rc_btn.h = 30;
-        rc_btn.x += 6;
-        rc_btn.w -= 12;
+    static rect_ui16_t get_radio_button_size() {                                // cannot be const(expr)
```
Comment (RadekVana) [inline src/gui/dialogs/DialogStateful.hpp:78]: I'm still not happy with removing constexpr (bigger code, less efficient)
If you insist, at least put definition of the method into cpp file.
And comment it with todo or fixme to be changed to constexpr after gui_defaults refactoring

**Pull request merged**